collector/accelerators_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+)
+
+type testAcceleratorCollector struct {
+	xc Collector
+}
+
+func (c testAcceleratorCollector) Collect(ch chan<- prometheus.Metric) {
+	c.xc.Update(ch)
+}
+
+func (c testAcceleratorCollector) Describe(ch chan<- *prometheus.Desc) {
+	prometheus.DescribeByCollect(c, ch)
+}
+
+func TestAccelerator(t *testing.T) {
+	testcase := `# HELP node_accelerator_cards Number of accelerator cards per type
+	# TYPE node_accelerator_cards counter
+	node_accelerator_cards{model="A100",vendor="NVIDIA"} 1
+	node_accelerator_cards{model="RTX_4090",vendor="NVIDIA"} 1
```

Comment:
(nit) can you change the fixture so it reports 2 here? just to check that we're adding up.

Reply:
Fixed